### DISCUSSION TITLE
Some questions about Indexing for Search

### DISCUSSION THREAD

**Discussion by LemonLiTree** [google/s2geometry#324] [Q&A]
I use encode data in our system to store S2 data, for example：
```c++
        std::string* buf;
        Encoder encoder;
        _polygon->Encode(&encoder);//_polygon is  std::unique_ptr<S2Polygon> _polygon; 
        buf->append(encoder.base(), encoder.length());
```
In the document introduction, support Indexing for Search 
http://s2geometry.io/devguide/cpp/quickstart
Can I build an index on such a data store? If so, is there a sample code that I can refer to, and if that doesn't work, how do I store the data so I can build an index.
Or is there more information about Indexing for Search  reference

**Comment by smcallis** [comment]
You can't index on the encoded data directly, but S2ShapeIndex is designed so that the index and shapes are encoded separately.   [Here's](https://github.com/google/s2geometry/blob/master/src/s2/encoded_s2shape_index.h#L80-L94) some examples in EncodedS2ShapeIndex showing how you can add shapes to an index an then encode them and the index separately.

**Reply by LemonLiTree** [reply]
If my data only involves calculations based on s2index, can I just store EncodedS2ShapeIndex data？ Or store encode shapes and encode index separately？

**Reply by smcallis** [reply]
You'd encode the shapes and index together in that case most likely, then just read the data into an EncodedS2ShapeIndex with a lazily decode [shape factory](https://github.com/google/s2geometry/blob/master/src/s2/s2shapeutil_coding.h#L176) (usually, you can do a full decode if you want but it's slower since you likely don't need all the shapes).

**Comment by LemonLiTree** [comment]
Is there any more documentation about Indexing for Search, I want to do a quick query based on a large amount of S2 region data

**Reply by smcallis** [reply]
I don't think we have anything comprehensive written down.  What kind of searches do you want to do?

**Reply by LemonLiTree** [reply]
just like this
<img width="875" alt="image" src="https://github.com/google/s2geometry/assets/109049295/1bc4ac2d-9dfe-4634-8e1d-cf76dd3e6284">

**Reply by smcallis** [reply]
Ah I'd reference the comments in that file showing how to use it: https://github.com/google/s2geometry/blob/master/src/s2/s2region_term_indexer.h#L1-L95
